src/joystick/hidapi/SDL_hidapi_steam.c

```diff
@@ -1197,11 +1235,24 @@ static bool HIDAPI_DriverSteam_UpdateDevice(SDL_HIDAPI_Device *device)
             }
 
             ctx->m_last_state = ctx->m_state;
+        } else if (joystick && D0G_IS_WIRELESS_DISCONNECT(pPacket, nPacketLength)) {
+            // Controller has disconnected from the wireless dongle
+            HIDAPI_JoystickDisconnected(device, device->joysticks[0]);
+            joystick = NULL;
+        } else if (!joystick && D0G_IS_WIRELESS_CONNECT(pPacket, nPacketLength)) {
+            // Controller has connected to the wireless dongle
+            if (!HIDAPI_JoystickConnected(device, NULL)) {
+                return false;
+            }
+
+            joystick = SDL_GetJoystickFromID(device->joysticks[0]);
```

Comment:
This probably isn't correct for multiple Steam Controllers paired to the same dongle.

Reply:
I think it should be right. According to the [comments in the Linux driver](https://github.com/torvalds/linux/blame/0a9b9d17f3a781dea03baca01c835deaa07f7cc3/drivers/hid/hid-steam.c#L1079-L1089), there's a separate interface enumerated for each potential controller that can be connected to the dongle. That was what was causing all those phantom controllers to appear that didn't actually produce any input.